Source/platform/locale.cpp

```diff
@@ -15,13 +15,42 @@
 #include <windows.h>
 #include <winnls.h>
 // clang-format on
-
-#include "utils/stdcompat/algorithm.hpp"
+#elif defined(__APPLE__)
+#include <CoreFoundation/CoreFoundation.h>
 #else
 #include <locale>
 #endif
 
+#include "utils/stdcompat/algorithm.hpp"
+#include "utils/stdcompat/string_view.hpp"
+
 namespace devilution {
+namespace {
+
+std::string IetfToPosix(string_view langCode)
+{
+	/*
+	 * Handle special case for simplified/traditional Chinese. IETF/BCP-47 specifies that only the script should be
+	 *  used to discriminate languages when the region doesn't add additional value. For chinese scripts zh-Hans is
+	 *  preferred over zh-Hans-CN (but platforms may include the region identifier anyway). POSIX locales don't use
+	 *  script in the same way so we need to convert these back to lang_region.
+	 */
+	if (langCode.substr(0, 7) == "zh-Hans") {

```

Comment:
starts_with

Reply:
That was introduced in C++20 unfortunately, doesn't look like we have a polyfill